echidna/core/smear.py

```diff
@@ -233,6 +237,81 @@ def random_gaussian_energy_spectra(self, true_spectrum):
         smeared_spectrum._raw_events = raw_events
         return smeared_spectrum
 
+
+    def random_gaussian_energy_spectra_parallel(self, true_spectrum,nProcs):
+        """ Smears the energy of a spectra object using multiprocessing  by generating
+          a number of random points from a Gaussian pdf generated
+          for that bin. The number of points generated is equivalent
+          to the number of entries in that bin.
+
+        Args:
+          true_spectrum (spectra): spectrum to be smeared
+
+        Returns:
+          A smeared spectra object.
+        """
+        def Worker(true_spectrum,lower,upper,out_q):
```

Comment:
I have a few comments/concerns about the `Worker` function (which I think relate to all the `Worker` functions defined):
1. Isn't the `Worker` function doing exactly the same thing as the `random_gaussian_energy_spectrum` method? Why could we not use the existing method and add  `lower`, `upper` and `out_q` as optional args? This would save a lot of repeated code and mean you could use the same method for both cases.
2. If we do have to use the `Worker` functions, why are they defined inside class methods?
3. Also if we do have to retain the `Worker` functions, can we please change the naming to `worker` . Please try to use `lower_case_and_underscores` for function, method and variable names, where possible, and only use `UpperCamelCase` for classes.